cookiecutter/repository.py

```diff
@@ -76,22 +76,27 @@ def determine_repo_dir(template, abbreviations, clone_to_dir, checkout,
     """
     template = expand_abbreviations(template, abbreviations)
 
+    repository_candidates = []
+
     if is_repo_url(template):
-        repo_dir = clone(
+        cloned_repo = clone(
             repo_url=template,
             checkout=checkout,
             clone_to_dir=clone_to_dir,
             no_input=no_input,
         )
-    else:
-        # If it's a local repo, no need to clone or copy to your
-        # cookiecutters_dir
-        repo_dir = template
+        repository_candidates.append(cloned_repo)
+
+    repository_candidates.extend([
+        template,
+        os.path.join(clone_to_dir, template)
+    ])
 
-    if repository_has_cookiecutter_json(repo_dir):
-        return repo_dir
+    for repo_candidate in repository_candidates:
+        if repository_has_cookiecutter_json(repo_candidate):
+            return repo_candidate
 
     raise RepositoryNotFound(
         'The repository {} could not be located or does not contain '
-        'a "cookiecutter.json" file.'.format(repo_dir)
+        'a "cookiecutter.json" file.'.format(template)
```

Comment:
It would be cool to extend this error message and list all the directories that have been checked, which would be the items in ``repository_candidates``.